starclass/todolist.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+
+.. codeauthor:: Rasmus Handberg <rasmush@phys.au.dk>
+"""
+
+import os
+import re
+import sqlite3
+from contextlib import closing
+from .io import load_lightcurve
+
+#--------------------------------------------------------------------------------------------------
+def create_fake_todolist(input_folder, output_todo='todo.sqlite', file_pattern=None,
+	overwrite=False):
+	"""
+
+	.. codeauthor:: Rasmus Handberg <rasmush@phys.au.dk>
+	"""
+
+	# Basic checks of the input:
+	if not os.path.isdir(input_folder):
+		raise NotADirectoryError(input_folder)
+	if file_pattern is None:
+		file_pattern = r'\.fits(\.gz)?$'
+
+	# Go through the input directory recursively and find all files which match the pattern:
+	files = []
+	regex = re.compile(file_pattern)
+	for root, dirnames, filenames in os.walk(input_folder, followlinks=True):
+		for filename in filenames:
+			if regex.match(filename):
```

Comment:
Change to `regex.search(filename)`.
`match` only checks beginning of the string apparently: https://docs.python.org/3/library/re.html#re.match

Reply:
Fixed in latest version